### PR TITLE
feat(plugin): migrate to sandy, improve performance

### PR DESCRIPTION
**Pull request by DenisIrkhin**
First and foremost: Thank You for the plugin! We started using it back in March 2021 and it is a super nice tool and the only possible Flipper plug-in back in the time. I highly doubt that we could start our RQ journey without the plugin.
Thank You, Bruno!

Motivation:
We started using RQ actively for React Native app. As we add more RQ based features we started to see the losing performance of the plugin. Now we have about 100 queries on average per user.
In particular, when we re-select a query it takes 2-3 sec to show the right sidebar.

Rendering the whole query tree in the DOM is a heavy task. So the idea was to see it collapse by default.
I tried to adjust the plug-in but it turned out that Flipper requires a new Sandy API and I have to migrate the plugin there.
After it's done we have a nice UI with dark mode and collapsed by default sidebar

Another improvement: we used to send the whole QueryCache on each query/observer change which is heavy for Flipper. Now we are sending only 

Description
- migrate plug-ins to Sandy https://fbflipper.com/docs/extending/sandy-migration/
- improve performance by setting collapsed tree for `Query Explorer` panel which tried hard to render in the DOM very large and complex query object
- add more columns to QueryTable
- improve QueryTable styles: now queryHash column has multiple lines
- move husky and linting checks in the monorepo root
- update dependencies
- implement an incremental update of QueryCache

### CODE REVIEW

**Comment by DenisIrkhin** [comment]
I've tested for our QueryCache flow - it works decently.
With intense Cache updates(as I tested) we have table scrolling to the bottom constantly. I added enableAutoScroll and it looks better but probably it could be done in a better way.

It's ready for review @bgaleotti Thanks in advance.

File: packages/react-query-native-devtools/package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "react-query-native-devtools",
   "author": "Bruno Galeotti",
-  "version": "3.0.1",
+  "version": "3.0.3",
```
[the remark this answers is not in the document]
Comment (DenisIrkhin) [reply packages/react-query-native-devtools/package.json]: @bgaleotti Thanks! Updated.

File: packages/flipper-plugin-react-query-native-devtools/src/index.tsx [hunk 2]
```diff
@@ -1,73 +1,179 @@
...
+        break;
+      case 'queryUpdated':
+      case 'observerAdded':
+      case 'observerRemoved':
+      case 'observerResultsUpdated':
+        queries.upsert(extendQuery(query));
+
+        // To re-render the sidebar when we have selected query and the query is updated
+        if (selectedQueryId.get()?.slice(11) === queryHash) {
+          selectedQueryId.set(makeQuerySelectionKey(query));
```
Comment (DenisIrkhin) [inline packages/flipper-plugin-react-query-native-devtools/src/index.tsx:97]: We need to change `selectedQueryId` when it's updated. The comparison happens by value. Even I tried to undefined it and set some intermediate values to selectedQueryId at the end of the day React part does not react on it. The final result set here is compared by value is QuerySidebar and if the value is the same than we have no re-render. I tried to strike re-render with standard React ways: by changing props, state and children but it did Not bring fruits. Maybe we can come up with the better solution, but this one works Ok.

**Comment by bgaleotti** [comment]
Released on v4.0.0 🚀.

Thanks for your contribution @DenisIrkhin 🙌!

**Pull request merged**